src/renderer/views/pool/PoolDetailsView.tsx

```diff
@@ -107,28 +101,39 @@ export const PoolDetailsView: React.FC = () => {
         <RefreshButton clickHandler={onRefreshData} disabled={refreshButtonDisabled} />
       </Styled.ControlsContainer>
       {FP.pipe(
-        RD.combine(routeAssetRD, poolDetailRD),
-        RD.fold(
-          renderInitialView,
-          () => <PoolDetails {...prevProps.current} isLoading />,
-          (e: Error) => {
-            return <PoolStatus label={intl.formatMessage({ id: 'common.error' })} displayValue={e.message} />
-          },
-          ([asset, poolDetail]) => {
-            prevProps.current = {
-              asset: O.some(asset),
-              depth: getDepth(poolDetail, priceRatio),
-              volume24hr: get24hrVolume(poolDetail, priceRatio),
-              allTimeVolume: getAllTimeVolume(poolDetail, priceRatio),
-              totalSwaps: 0 /* getTotalSwaps(history) */,
-              totalStakers: 0 /* getTotalStakers(poolDetail) */,
-              priceUSD: getPriceUSD(poolDetail, priceRatio),
-              priceSymbol: O.toUndefined(priceSymbol),
-              HistoryView: PoolHistory
-            }
-
-            return <PoolDetails {...prevProps.current} />
-          }
+        oRouteAsset,
+        O.fold(
+          () => (
+            <PoolStatus
+              label={intl.formatMessage({ id: 'common.error' })}
+              displayValue={intl.formatMessage({ id: 'routes.invalid.asset' }, { asset })}
+            />
```

Comment:
You can use `ErrorView` here (like in other views for same purpose) and `PoolStatus` might be not needed anymore (and can be removed?).

Reply:
yes. changed to use `ErrorView` but can not remove it as PoolStatus used at another place too
![image](https://user-images.githubusercontent.com/67793194/113327532-c13beb00-9323-11eb-9913-81c1dacec640.png)

Reply:
Ok